### PR TITLE
Session1 天気予報アプリの画面レイアウトを構成する

### LINKED ISSUE #1: Session0
# 課題の準備をする

課題をスムーズに進めるための準備をしましょう。

## 課題

- [x] 作成した研修用のリポジトリにメンターをコラボレーターとして招待する
- [x] 作成した研修用のリポジトリの GitHub Actions の権限を「Allow all actions and reusable workflows」に設定
- [x] 作成した研修用のリポジトリの GitHub Actions のワークフローの権限を「Read and write permissions」に設定
- [x] 作成した研修用のリポジトリでプルリクエストマージ後に自動でヘッドブランチが削除されるように設定
- [x] 作成した研修用のリポジトリにルールセットを作成
  - [x] 対象のブランチはデフォルトブランチ
  - [x] 削除を許可しない
  - [x] マージ前にプルリクエスト必須
    - [x] マージ前に１人の承認を必須
    - [x] マージ前に会話の解決必須
  - [x] フォースプッシュを許可しない
- [x] 使用しているチャットツールで GitHub 通知の購読設定
  - Slackコマンド:
    - `/github subscribe ユーザー名/リポジトリ名 issues pulls reviews comments`
    - `/github unsubscribe ユーザー名/リポジトリ名 commits releases deployments workflows branches discussions`
- [x] 作成した研修用のリポジトリを `git clone [url]` して、ローカルにコピー
- [x] [fvm] をインストール
- [x] プロジェクトルートで `fvm install` コマンドを実行して対象の Flutter SDK をインストール
- [x] IDE の最新安定板をインストール
- [x] Flutter の設定（ `fvm flutter doctor` を実行して確認）
- [x] Android・iOS でアプリを実行
- [x] `.github/workflows-templates` ディレクトリを `.github/workflows` にリネーム
- [x] [review-assign.yaml] の `REVIEWERS` にメンターの GitHub のユーザー名を設定
- [x] 作成した研修用のリポジトリのルールセットを編集（上２つの修正を行ったプルリクエストでステータスチェック完了後に対応する）
  - [x] マージ前にステータスチェック必須
    - [x] マージ前にブランチ最新化必須
    - [x] ステータスチェック `flutter test` `check` を追加

## 注意事項

- `yumemi_lints` が Flutter SDK の最新バージョンをサポートおらずエラーが発生してしまう場合があります。詳しくは [Error occurs when yumemi_lints does not support the latest version of Flutter SDK] をご覧ください。

## 参考資料

- [コラボレーターを個人リポジトリに招待する]
- [リポジトリの GitHub Actions の設定を管理する]
- [ルールセットについて]
- [ステータスチェックについて]
- [GitHub + Slack]

[fvm]: https://fvm.app/

[review-assign.yaml]: https://github.com/yumemi-inc/flutter-training-template/blob/main/.github/templates/.github/workflows-templates/review-assign.yaml#L8

[Error occurs when yumemi_lints does not support the latest version of Flutter SDK]: https://github.com/yumemi-inc/flutter-training-template/wiki/Error-occurs-when-yumemi_lints-does-not-support-the-latest-version-of-Flutter-SDK

[コラボレーターを個人リポジトリに招待する]: https://docs.github.com/ja/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository#inviting-a-collaborator-to-a-personal-repository

[リポジトリの GitHub Actions の設定を管理する]: https://docs.github.com/ja/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#allowing-select-actions-and-reusable-workflows-to-run

[ルールセットについて]: https://docs.github.com/ja/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets

[ステータスチェックについて]: https://docs.github.com/ja/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks

[GitHub + Slack]: https://slack.github.com

### LINKED ISSUE #2: Session1
# 天気予報アプリの画面レイアウトを構成する

## 課題

以下の条件の画面レイアウトを作ってみましょう。

- [Placeholder] の幅は画面の幅の半分
- 青字と赤字の [Text] の幅は [Placeholder] の幅の半分
  ![horizontal-ratio]

- [Placeholder] の高さと幅は同じ
  ![aspect-ratio]

- [Text] の上下に 16 logical pixel のパディング
- [Text] の文字は `** ℃`
- [Text] の水平位置は中央
- [Text] のスタイルは `Theme.of(context).textTheme` で取得してきた `TextTheme` の [`labelLarge`]
- [Text] の色は左が [`Colors.blue`]、右が [`Colors.red`]
  ![text-padding]

- [Placeholder] の水平中央は画面の中央と同じ
- [Placeholder] と [Text] を合わせた矩形の垂直中央は画面の中央と同じ
  ![centering]

- [Text] と [TextButton] の隙間は 80 logical pixel
  ![button-margin]

- [Text] と [TextButton] の水平中央は同じ
  ![button-centering]

## ヒント

Flutter には iOS の [NSLayoutConstraint] や Android の [ConstraintLayout] のようなものはありません。

以下の Widget を利用するとさまざまな画面に対応することができます。

- [Row]
- [Column]
- [Flexible]
- [Expanded]
- [AspectRatio]
- [FractionallySizedBox]

[Flutter Inspector] を利用すると、レイアウト構造を視覚化でき、より開発しやすくなります。
便利な機能がたくさんあるので、実際に試しながらしっかり読み込んでおきましょう。

[horizontal-ratio]: https://github.com/yumemi-inc/flutter-training-template/blob/main/docs/sessions/images/layout/horizontal-ratio.png?raw=true

[aspect-ratio]: https://github.com/yumemi-inc/flutter-training-template/blob/main/docs/sessions/images/layout/aspect-ratio.png?raw=true

[text-padding]: https://github.com/yumemi-inc/flutter-training-template/blob/main/docs/sessions/images/layout/text-padding.png?raw=true

[centering]: https://github.com/yumemi-inc/flutter-training-template/blob/main/docs/sessions/images/layout/centering.png?raw=true

[button-margin]: https://github.com/yumemi-inc/flutter-training-template/blob/main/docs/sessions/images/layout/button-margin.png?raw=true

[button-centering]: https://github.com/yumemi-inc/flutter-training-template/blob/main/docs/sessions/images/layout/button-centering.png?raw=true

[Placeholder]: https://api.flutter.dev/flutter/widgets/Placeholder-class.html

[Text]: https://api.flutter.dev/flutter/widgets/Text-class.html

[`labelLarge`]: https://api.flutter.dev/flutter/material/TextTheme/labelLarge.html

[`Colors.blue`]: https://api.flutter.dev/flutter/material/Colors/blue-constant.html

[`Colors.red`]: https://api.flutter.dev/flutter/material/Colors/red-constant.html

[TextButton]: https://api.flutter.dev/flutter/material/TextButton-class.html

[NSLayoutConstraint]: https://developer.apple.com/documentation/uikit/nslayoutconstraint

[ConstraintLayout]: https://developer.android.com/develop/ui/views/layout/constraint-layout

[Row]: https://api.flutter.dev/flutter/widgets/Row-class.html

[Column]: https://api.flutter.dev/flutter/widgets/Column-class.html

[Flexible]: https://api.flutter.dev/flutter/widgets/Flexible-class.html

[Expanded]: https://api.flutter.dev/flutter/widgets/Expanded-class.html

[AspectRatio]: https://api.flutter.dev/flutter/widgets/AspectRatio-class.html

[FractionallySizedBox]: https://api.flutter.dev/flutter/widgets/FractionallySizedBox-class.html

[Flutter Inspector]: https://docs.flutter.dev/development/tools/devtools/inspector

### PR DESCRIPTION
**Pull request by yumemi-rnose**
## 課題
天気予報アプリの画面レイアウトを構成する
close #2 

## 対応箇所

- 画面レイアウト
  - [x] [Placeholder](https://api.flutter.dev/flutter/widgets/Placeholder-class.html) の幅は画面の幅の半分
  - [x] 青字と赤字の [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) の幅は [Placeholder](https://api.flutter.dev/flutter/widgets/Placeholder-class.html) の幅の半分
  - [x] [Placeholder](https://api.flutter.dev/flutter/widgets/Placeholder-class.html) の高さと幅は同じ
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) の上下に 16 logical pixel のパディング
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) の文字は ** ℃
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) の水平位置は中央
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) のスタイルは Theme.of(context).textTheme で取得してきた TextTheme の [labelLarge](https://api.flutter.dev/flutter/material/TextTheme/labelLarge.html)
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) の色は左が [Colors.blue](https://api.flutter.dev/flutter/material/Colors/blue-constant.html)、右が [Colors.red](https://api.flutter.dev/flutter/material/Colors/red-constant.html)
  - [x] [Placeholder](https://api.flutter.dev/flutter/widgets/Placeholder-class.html) の水平中央は画面の中央と同じ
  - [x] [Placeholder](https://api.flutter.dev/flutter/widgets/Placeholder-class.html) と [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) を合わせた矩形の垂直中央は画面の中央と同じ
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) と [TextButton](https://api.flutter.dev/flutter/material/TextButton-class.html) の隙間は 80 logical pixel
  - [x] [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) と [TextButton](https://api.flutter.dev/flutter/material/TextButton-class.html) の水平中央は同じ

## 動作

| iOS | Android |
|----------|--------|
| ファイル名 | ファイル名 |

### CODE REVIEW

File: lib/main_screen.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    final textStyle = Theme.of(context).textTheme.labelLarge;
+    return Scaffold(
+        body: Center(
+          child: FractionallySizedBox(
+            widthFactor: 0.5,
+            child: Column(
+              children: [
+                Expanded(
+                  child: Container(),
+                ),
```
Comment (nerd0geek1) [inline lib/main_screen.dart]: @yumemi-rnose 
`Expanded`を上下に設定するのは :cool: ですが、画面上部側に設定すべき子要素がないのであれば、`Spacer` Widgetを設定したほうが良いですね。
https://api.flutter.dev/flutter/widgets/Spacer-class.html

**Review by nerd0geek1 — COMMENTED** [review]
@yumemi-rnose 
1点フィードバックした点を除いてLGTMです。

File: lib/main_screen.dart (hunk 1 above)
Comment (yumemi-rnose) [reply lib/main_screen.dart]: @nerd0geek1 
レビューありがとうございます。修正しました。ご確認お願いします。
[a128535](https://github.com/yumemi-rnose/yumemi-flutter-training/pull/14/commits/a128535e46daa67d3a03f0a29471029ce88c9ae4)

**Review by nerd0geek1 — APPROVED** [review]
LGTM

**Pull request merged**